lib/ts/recipe/passwordless/emaildelivery/services/backwardCompatibility/index.ts

```diff
@@ -0,0 +1,100 @@
+        preAuthSessionId: string;
+    },
+    __: any
+): Promise<void> {}
+
+export default class BackwardCompatibilityService
+    implements EmailDeliveryInterface<TypePasswordlessEmailDeliveryTypeInput> {
+    private isInServerlessEnv: boolean;
+    private createAndSendCustomEmail: (
+        input: {
+            // Where the message should be delivered.
+            email: string;
+            // This has to be entered on the starting device  to finish sign in/up
+            userInputCode?: string;
+            // Full url that the end-user can click to finish sign in/up
+            urlWithLinkCode?: string;
+            codeLifetime: number;
+            // Unlikely, but someone could display this (or a derived thing) to identify the device
+            preAuthSessionId: string;
+        },
+        userContext: any
+    ) => Promise<void>;
+
+    constructor(
+        isInServerlessEnv: boolean,
+        createAndSendCustomEmail?: (
+            input: {
+                // Where the message should be delivered.
+                email: string;
+                // This has to be entered on the starting device  to finish sign in/up
+                userInputCode?: string;
+                // Full url that the end-user can click to finish sign in/up
+                urlWithLinkCode?: string;
+                codeLifetime: number;
+                // Unlikely, but someone could display this (or a derived thing) to identify the device
+                preAuthSessionId: string;
+            },
+            userContext: any
+        ) => Promise<void>
+    ) {
+        this.isInServerlessEnv = isInServerlessEnv;
+        this.createAndSendCustomEmail =
+            createAndSendCustomEmail === undefined ? defaultCreateAndSendCustomEmail : createAndSendCustomEmail;
+    }
+
+    sendEmail = async (input: TypePasswordlessEmailDeliveryTypeInput & { userContext: any }) => {
+        try {
+            if (!this.isInServerlessEnv) {
+                this.createAndSendCustomEmail(
+                    {
+                        email: input.email,
+                        userInputCode: input.userInputCode,
+                        urlWithLinkCode: input.urlWithLinkCode,
+                        preAuthSessionId: input.preAuthSessionId,
+                        codeLifetime: input.codeLifetime,
+                    },
+                    input.userContext
+                );
+            } else {
+                await this.createAndSendCustomEmail(
+                    {
+                        email: input.email,
+                        userInputCode: input.userInputCode,
+                        urlWithLinkCode: input.urlWithLinkCode,
+                        preAuthSessionId: input.preAuthSessionId,
+                        codeLifetime: input.codeLifetime,
+                    },
+                    input.userContext
+                );
+            }
+        } catch (_) {}
```

Comment:
In a previous comment, you asked if `isServerlessEnv` should be considered or not. The answer is no, it shouldn't. Cause in other implementations, we are doing it in the "background" (without await) in case `!this.isInServerlessEnv`, so that the API returns quickly. But this prevents us from propagating errors from the email sending service to the end user.

So instead, we always await it now. Since we always await it, we don't need to check for `isServerlessEnv` anymore.